src/librustc/middle/lib_features.rs

```diff
@@ -143,6 +144,26 @@ impl<'a, 'tcx> Visitor<'tcx> for LibFeatureCollector<'a, 'tcx> {
         NestedVisitorMap::All(&self.tcx.hir)
     }
 
+    fn visit_item(&mut self, item: &'tcx Item) {
+        match item.node {
+            ItemKind::ExternCrate(_) => {
```

Comment:
What about Rust 2018, where `extern crate` is optional? Do inject these statements into HIR? That is, can we rely on them being present even if they come just from `Cargo.toml`?

Reply:
That's a good point; I don't think this will work in the 2018 edition. Let me try tweaking things again.